### ISSUE TITLE
Gabled Roofs

### ISSUE THREAD

**Issue by guillemc23** [Luvideria/mcfgJSONtoOBJ#3]
Would it be possible to export roofs as gabled roof instead of flat ones? I took a look at the code but I can't find any parameter to do so :/

**Comment by guillemc23** [comment]
Love this tool btw, exactly what I was looking for... I even thought of writing it myself 👍🏻

**Comment by Luvideria** [comment]
There is no parameter for this yet. I am thinking of a procedure to make them in a reliable manner. It's easy for square houses, but for L houses, or U houses, it's a bit more tricky. There are also triangular houses, but those are easy.
We need to detect the case somehow. I thought about using the angle between each subsequent vertex. If I remember correctly, vertices are always in the same clockwise or anti clockwise order.
Technically, L houses are 2 rectangles and U three. Maybe just putting the roofs intersecting each others would be a good first shot.

**Comment by guillemc23** [comment]
I gave it a try because I thought many of them would be rectangles, and it might have been a good idea to look for the shortest side's center and then build the roof across the long side direction... but when I took a closer look at the length of each contour I found some weird 6, 7 and even a 14 sided buildings that made me think that it wasn't as straightforward as I thought it was 😅

**Comment by Luvideria** [comment]
Yes, you can refer to the jupyter notebook create_building.ipynb. I display a graph of all buildings, and the diversity of shapes is quite telling.
